.github/workflows/build-linux.yml

```diff
@@ -120,31 +119,43 @@ jobs:
           repository: fragcolor-xyz/shards
           fetch-depth: 2
           submodules: ${{ steps.setup.outputs.submodules }}
+      - name: Download artifact (Release)
+        if: ${{ needs.Linux.outputs.build-type == 'Release' }}
+        uses: actions/download-artifact@v3
+        with:
+          name: shards-linux ${{ needs.Linux.outputs.build-type }}
+          path: build
       - name: Set up dependencies (Debug)
-        if: ${{ steps.setup.outputs.build-type == 'Debug' }}
+        if: ${{ needs.Linux.outputs.build-type == 'Debug' }}
         run: |
           sudo apt-get -y update
           sudo apt-get -y install build-essential git cmake wget clang ninja-build xorg-dev libdbus-1-dev libssl-dev lcov mesa-utils
           ./bootstrap
           rustup toolchain install nightly
           rustup default nightly
       - uses: Swatinem/rust-cache@v2
-        if: ${{ steps.setup.outputs.rust-cache == 'true' }}
+        if: ${{ needs.Linux.outputs.rust-cache == 'true' && needs.Linux.outputs.build-type == 'Debug' }}
         with:
-          key: ${{ steps.setup.outputs.build-type }}
+          key: ${{ needs.Linux.outputs.build-type }}
       - name: Build (Debug)
-        if: ${{ steps.setup.outputs.build-type == 'Debug' }}
+        if: ${{ needs.Linux.outputs.build-type == 'Debug' }}
         run: |
-          mkdir build
+          if [ ! -d ./build ]
+          then
+            mkdir build
+          fi
           cd build
-          cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCODE_COVERAGE=1 ..
-          ninja shards
-      - name: Download artifact (Release)
-        if: ${{ steps.setup.outputs.build-type == 'Release' }}
-        uses: actions/download-artifact@v3
-        with:
-          name: shards-linux ${{ steps.setup.outputs.build-type }}
-          path: build
+
+          cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCODE_COVERAGE=1 -DSHARDS_BUILD_TESTS=1 ..
+
+          if [ "${{ needs.Linux.outputs.run-tests }}" == "true" ]
+          then
+            ninja shards
+          fi
+          if [ "${{ needs.Linux.outputs.run-extra-tests }}" == "true" ]
+          then
+            ninja test-runtime
```

Comment:
I would even just remove `run-extra-tests` and just run this under `run-tests`, no point in disabling only test-runtime. It's technically not even testing any of the 'extra' shards.

Reply:
Ok. It was separate before so I thought I had to keep it that way. But yeah let's test everything.